pkg/apis/hco/v1beta1/hyperconverged_types.go

```diff
@@ -56,25 +57,31 @@ type HyperConvergedConfig struct {
 	NodePlacement *sdkapi.NodePlacement `json:"nodePlacement,omitempty"`
 }
 
-type HyperConvergedFeatureGates map[string]bool
-
-func (fgs HyperConvergedFeatureGates) IsEnabled(fgName string) bool {
-	fg, found := fgs[fgName]
-	return fg && found
+// HyperConvergedFeatureGates is a set of optional feature gates to enable or disable new features that are not enabled
+// by default yet.
+// +optional
+// +k8s:openapi-gen=true
+type HyperConvergedFeatureGates struct {
+	// Allow attaching a data volume to a running VMI
+	// +optional
+	HotplugVolumes *bool `json:"HotplugVolumes,omitempty"`
 }
 
 // get list of feature gates from a specific operand list
-func (fgs HyperConvergedFeatureGates) GetFeatureGateList(candidates []string) []string {
-	res := make([]string, 0, len(fgs))
-	for _, fg := range candidates {
-		if fgs.IsEnabled(fg) {
-			res = append(res, fg)
-		}
+func (fgs *HyperConvergedFeatureGates) GetFeatureGateList() []string {
```

Comment:
I personally don't think this function is that needed (see other comments), but if keeping it, and since it has a pointer receiver, maybe support the case in which `fgs == nil`, and return an empty slice (instead of panic as now).

Reply:
Needed for the configmap. butyes, we can live without it.